akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/PersistentBehaviorImpl.scala

```diff
@@ -25,32 +25,39 @@ private[akka] final case class PersistentBehaviorImpl[Command, Event, State](
   tagger:            Event ⇒ Set[String]                                       = (_: Event) ⇒ Set.empty[String],
   snapshotWhen:      (State, Event, Long) ⇒ Boolean                            = ConstantFun.scalaAnyThreeToFalse,
   recovery:          Recovery                                                  = Recovery()
-) extends PersistentBehavior[Command, Event, State] {
+) extends PersistentBehavior[Command, Event, State] with EventsourcedStashReferenceManagement {
 
   override def apply(context: typed.ActorContext[Command]): Behavior[Command] = {
     Behaviors.setup[EventsourcedBehavior.InternalProtocol] { ctx ⇒
       Behaviors.withTimers { timers ⇒
-        val settings = EventsourcedSettings(ctx.system)
-          .withJournalPluginId(journalPluginId.getOrElse(""))
-          .withSnapshotPluginId(snapshotPluginId.getOrElse(""))
-        val setup = new EventsourcedSetup(
-          ctx,
-          timers,
-          persistenceId,
-          initialState,
-          commandHandler,
-          eventHandler,
-          WriterIdentity.newIdentity(),
-          recoveryCompleted,
-          tagger,
-          snapshotWhen,
-          recovery,
-          holdingRecoveryPermit = false,
-          settings = settings,
-          internalStash = StashBuffer(settings.stashCapacity)
-        )
+        {
```

Comment:
What is this extra block?

Reply:
Online merging is evil :sweat:. Noted for the future and removed redundant block.